app/controllers/posts_controller.rb

```diff
@@ -1,16 +1,35 @@
 class PostsController < ApplicationController
-  before_action :set_user
-
   def index
-    @posts = Post.all
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.order(created_at: :asc)
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      flash[:notice] = 'Post created successfully!'
+      redirect_to user_posts_path(current_user)
+    else
+      flash[:alert] = "Couln't create post!"
+      render :new, status: :unprocessable_entity
+    end
```

Comment:
### When tried to create a post
![Screenshot from 2023-03-05 03-05-36](https://user-images.githubusercontent.com/26767383/222930232-11c55b58-751d-4dae-95be-6fe7e41829af.png)
![Screenshot from 2023-03-05 03-09-13](https://user-images.githubusercontent.com/26767383/222930270-5dcb4fb9-5e1f-42a8-891c-1fab00d17efa.png)